java/mtsj/core/src/main/java/com/devonfw/application/mtsj/general/common/base/TokenManagementService.java

```diff
@@ -0,0 +1,55 @@
+package com.devonfw.application.mtsj.general.common.base;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.stereotype.Component;
+
+import com.devonfw.application.mtsj.general.common.api.datatype.Role;
+import com.devonfw.application.mtsj.general.common.api.to.UserDetailsClientTo;
+import com.devonfw.module.security.jwt.common.api.JwtAuthenticator;
+import com.devonfw.module.security.jwt.common.base.JwtConstants;
+
+/**
+ * Retrieves principal and authorities from token
+ *
+ */
+@Component
+public class TokenManagementService {
+
+  @Inject
+  private JwtAuthenticator jwtAuthenticator;
+
+  /**
+   *
+   * Retrieves principal and authorities from token
+   *
+   * @param jwt
+   * @return
+   */
+  public UserDetailsClientTo getUserdetailsFromTokenJwt(String jwt) {
+
+    UserDetailsClientTo userDetails = new UserDetailsClientTo();
+
+    Authentication auth = this.jwtAuthenticator.authenticate(jwt.replace(JwtConstants.TOKEN_PREFIX, ""));
```

Comment:
aint this a duplicated authentication? what is the purpose of this service in MTSJ?
IMHO this is to retrieve `UserDetailsClientTo` via REST service on the client. If so the existing authentication filter should be used and the `UserDetailsClientTo` should be created from the spring-security `Authentication` rather than again authenticating.